File: backend/src/main/java/info/gamewise/lor/videos/domain/VideoRegion.java
package info.gamewise.lor.videos.domain;

import com.github.brunosc.lor.domain.LoRRegion;

public final class VideoRegion implements Comparable<VideoRegion> {
    private final String code;
    private final String description;

    public VideoRegion(LoRRegion region) {
        this.code = region.getCode();
        this.description = region.prettyName();
    }

    public String getCode() {
        return code;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public int compareTo(VideoRegion o) {
        return this.getCode().compareTo(o.getCode());
    }

    @Override
    public String toString() {
        return "VideoRegion{" +
                "code='" + code + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/domain/VideoRegion.kt
package info.gamewise.lor.videos.domain

import com.github.brunosc.lor.domain.LoRRegion

class VideoRegion(region: LoRRegion) : Comparable<VideoRegion> {
    val code: String
    val description: String

    init {
        this.code = region.code
        description = region.prettyName()
    }

    override fun compareTo(other: VideoRegion): Int {
        return this.code.compareTo(other.code)
    }

    override fun toString(): String {
        return "VideoRegion{" +
                "code='" + code + '\'' +
                ", description='" + description + '\'' +
                '}'
    }
}